src/atoms/Button/Button.native.js

```diff
@@ -1,35 +1,265 @@
+    }
+  },
+  underlayColor({ variant, variantColor, theme }) {
+    switch (variant) {
+      case 'primary':
+      default:
+        return getColor(theme, `${variantColor || 'primary'}.600`); //in design, "Exception" white is used.
+      case 'secondary':
+        return getColor(theme, `${variantColor || 'primary'}.950`);
+      case 'tertiary':
+        return getColor(theme, `${variantColor || 'primary'}.950`);
+    }
+  },
+  padding({ size, theme, text }) {
+    switch (size) {
+      case 'xsmall':
+        if (!text) {
+          return `${theme.spacings.xxsmall}`;
+        }
+        return `${theme.spacings.xxsmall} ${theme.spacings.small}`;
+      case 'small':
+        if (!text) {
+          return `${makePxValue(0.75)}`;
+        }
+        return `${makePxValue(0.625)} ${theme.spacings.large}`;
+      case 'medium':
+      default:
+        if (!text) {
+          return `${theme.spacings.small}`;
+        }
+        return `${theme.spacings.small} ${theme.spacings.xxlarge}`;
+      case 'large':
+        if (!text) {
+          return `${makePxValue(1.25)}`;
+        }
+        return `${makePxValue(1.25)} ${theme.spacings.xxlarge} `;
+    }
+  },
+  iconSize({ size, children }) {
+    switch (size) {
+      case 'xsmall':
+        if (!children) {
+          return 'small';
+        }
+        return 'xsmall';
+      case 'small':
+        if (!children) {
+          return 'medium';
+        }
+        return 'small';
+      case 'medium':
+      case 'large':
+      default:
+        if (!children) {
+          return 'large';
+        }
+        return 'medium';
+    }
+  },
+  fontSize({ size }) {
+    switch (size) {
+      case 'xsmall':
+        return 'xxsmall';
+      case 'small':
+        return 'xsmall';
+      case 'medium':
+      case 'large':
+      default:
+        return 'medium';
+    }
+  },
+  alignSelf({ align }) {
+    switch (align) {
+      case 'left':
+      default:
+        return 'flex-start';
+      case 'center':
+        return 'center';
+      case 'right':
+        return 'flex-end';
+    }
+  },
+  letterSpacing({ size }) {
+    switch (size) {
+      case 'xsmall':
+        return 'medium';
+      default:
+        return 'small';
+    }
+  },
+  spaceBetween({ size, iconPosition }) {
+    switch (size) {
+      case 'xsmall':
+      case 'small':
+        return iconPosition === 'left' ? [0, 0.5, 0, 0] : [0, 0, 0, 0.5];
+      case 'medium':
+      case 'large':
+      default:
+        return iconPosition === 'left' ? [0, 1, 0, 0] : [0, 0, 0, 1];
+    }
+  },
+};
 
-const StyledText = styled(Text)(
+const StyledButton = styled(TouchableHighlight)(
   (props) => `
-  color: ${props.theme.colors.tone[900]};
-  align-self: center;
-  font-size: 18px;
-  letter-spacing: ${props.theme.fonts.letterSpacing.l};
-`,
+    flex-direction: row;
+    align-items: center;
+    align-self: ${styles.alignSelf(props)};
+    background-color: ${styles.backgroundColor(props)};
```

Comment:
Don't pass props here, just do

```suggestion
    background-color: ${styles.backgroundColor};
```

after making the 
```
const StyledButton = styled(TouchableHighlight)`
```

change so that props are passed automatically